axelor-business-project/src/main/java/com/axelor/apps/businessproject/service/InvoiceServiceProjectImpl.java

```diff
@@ -117,4 +131,16 @@ public Invoice updateLines(Invoice invoice) {
     }
     return invoice;
   }
+
+  public String exportAnnex(Invoice invoice) throws AxelorException {
+    PrintingTemplate invoicingAnnexPrintTemplate =
+        appBusinessProjectService.getAppBusinessProject().getInvoicingAnnexPrintTemplate();
+    if (invoicingAnnexPrintTemplate == null) {
+      throw new AxelorException(
+          TraceBackRepository.CATEGORY_CONFIGURATION_ERROR,
+          I18n.get(BaseExceptionMessage.TEMPLATE_CONFIG_NOT_FOUND));
+    }
+    return printingTemplatePrintService.getPrintLink(
+        invoicingAnnexPrintTemplate, new PrintingGenFactoryContext(invoice));
+  }
```

Comment:
please either:
* add this new method in a new service, for example `InvoiceProjectPrintAnnexService.java`, 
* or add the check and the throw exception in a new method `getInvoicingAnnexPrintTemplate` in `appBusinessProjectService` and inject `PrintingTemplatePrintService` and `AppBusinessProjectService` directly in the controller